docs/guide/blog/comment-blog.md

```diff
@@ -0,0 +1,607 @@
+
+```go
+rpc CreateComment(MsgCreateComment) returns (MsgCreateCommentResponse);
+```
+
+Now, add the `id` field to `MsgCreatePost`: 
+
+```go
+message MsgCreatePost {
+  string creator = 1;
+  string title = 2;
+  string body = 3;
+  uint64 id = 4;
+}
+```
+
+## Process messages
+
+In the newly scaffolded `x/blog/keeper/msg_server_create_comment.go` file, you can see a placeholder implementation of the `CreateComment` function (marked with `//TODO`). Right now it does nothing and returns an empty response. For your blog chain, you want the contents of the message (title and body) to be written to the state as a new comment.
+
+You need to do the following things:
+
+* Create a variable of type `Comment` with title and body from the message
+* Check if the the comment posted for the respective blog id exists and comment is not older than 100 blocks
+* Append this `Comment` to the store
+
+```go
+import (
+    //...
+    sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (k msgServer) CreateComment(goCtx context.Context, msg *types.MsgCreateComment) (*types.MsgCreateCommentResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	post := k.GetPost(ctx, msg.PostID)
+	postId := post.Id
+
+	// Create variable of type comment
+	var comment = types.Comment{
+		Creator:   msg.Creator,
+		Id:        msg.Id,
+		Body:      msg.Body,
+		Title:     msg.Title,
+		PostID:    msg.PostID,
+		CreatedAt: ctx.BlockHeight(),
+	}
+
+	// Check if the Post Exists for which a comment is being created
+	if msg.PostID > postId {
+		return nil, sdkerrors.Wrapf(types.ErrID, "Post Blog Id does not exist for which comment with Blog Id %d was made", msg.PostID)
+	}
+
+	// Check if the comment is older than the Post. If more than 100 blocks, then return error.
+	if comment.CreatedAt > post.CreatedAt+100 {
+		return nil, sdkerrors.Wrapf(types.ErrCommentOld, "Comment created at %d is older than post created at %d", comment.CreatedAt, post.CreatedAt)
+	}
+
+	id := k.AppendComment(ctx, comment)
+	return &types.MsgCreateCommentResponse{Id: id}, nil
+}
+```
+
+When the Comment validity is checked, it throws 2 error messages - `ErrID` and `ErrCommendOld`. You can define the error messages by adding error definitions to `x/blog/types/errors.go`:
+
+```go
+//...
+var (
+	ErrCommentOld = sdkerrors.Register(ModuleName, 1300, "")
+	ErrID = sdkerrors.Register(ModuleName, 1400, "")
+)
+```
+
+
+In the existing `x/blog/keeper/msg_server_create_post.go` file, you need to make a modification to add `createdAt`
+
+```go
+func (k msgServer) CreatePost(goCtx context.Context, msg *types.MsgCreatePost) (*types.MsgCreatePostResponse, error) {
+	// Get the context
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	// Create variable of type Post
+	var post = types.Post{
+		Creator:   msg.Creator,
+		Id:        msg.Id,
+		Title:     msg.Title,
+		Body:      msg.Body,
+		CreatedAt: ctx.BlockHeight(),
```

Comment:
```
error while running command go: # github.com/cosmonaut/blog/x/blog/keeper
../../x/blog/keeper/msg_server_create_comment.go:24:3: cannot use ctx.BlockHeight() (type int64) as type uint64 in field value
```

When scaffolding a comment in the following command:

```
starport scaffold list comment --no-message creator:string title:string body:string postID:uint createdAt:int 
```

`createdAt` was specified as `int`, which is a shorthand for `int32`. `ctx.BlockHeight()` is `uint64`.

Reply:
Made changes, `starport scaffold list comment --no-message creator:string title:string body:string postID:uint createdAt:uint`

Testing again for consistency.